src/app/GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs

```diff
@@ -47,12 +50,16 @@ public void ClearCache()
 
         public string GetCurrentBranchName(string path)
         {
+            IGitModule module = _getModule(path);
+
             if (!AppSettings.ShowRepoCurrentBranch || GitModule.IsBareRepository(path))
             {
                 return string.Empty;
             }
 
-            return GitModule.GetSelectedBranchFast(path);
+            string branchName = module.GetSelectedBranch();
```

Comment:
With this change, a Git command could be required for many repos, like submodules where no branch is checked out.
I have 150 modules in my dashboard and 100 of them are submodules, so GE will require 10s extra or so.